### DISCUSSION TITLE
In the 'ignoreip' configuration can I put a partial DNS?

### DISCUSSION THREAD

**Discussion by ccheath** [fail2ban/fail2ban#3610] [Q&A] [answered]
I just found out that the 'ignoreip' configuration can have DNS hosts.
I would like to add a domain (comcast.net) for instance.
Will this work in whitelisting all hosts that resolve to a comcast.net hostname?

**Comment by sebres** [comment, marked as answer]
> Will this work in whitelisting all hosts that resolve to a comcast.net hostname?

The answer is vice versa - it would ignore all IP that belong to DNS (resolved forwards) - no reverse DNS takes place (it would be vulnerable, because an attacker could register for some IP a reverse DNS entry to that domain. Thus only forward resolve would work.
For everything else there is `ignorecommand` where one is able to define more complex ignore requirements.

**Reply by ccheath** [reply]
Sebres, 
Thanks for the clarification.
What I'm trying to do is whitelist/ignore my own connections.
(I recently blocked my home IP while doing some testing and it took me a few hours to figure out what had happened.)
My home IP will change from time to time (being residential DHCP).
Is there a suggested way to approach/solve this issue?

**Reply by sebres** [reply]
Then I don't follow how ignoring of your domain may help here (unless your dynamic IP belongs to it).

> Is there a suggested way to approach/solve this issue?

Not really, excepting something like DynDNS (usually the routers have a callback triggered if IP changes). 
Or some command that can check the IP is your dynamic IP (so it could be specified via `ignorecommand`).
See https://github.com/fail2ban/fail2ban/issues/3504#issuecomment-1520775361 for similar question.

**Reply by ccheath** [reply]
what i've done (short term) is to put my IP in  `/etc/hosts` as `subdomain.mydomain.com` and add that to the `ignoreip` config line
what i plan to do (long term) is to write a small bit of webserver code (php maybe) that i can hit from a cron job on a computer inside my home network (protected with an api key) that will then update the hosts file if the IP has changed

i found some [webmin/virtualmin users](https://forum.virtualmin.com/t/solution-for-ddns-server-on-virtualmin-proposal-to-add-command-modify-record-with-ttl-to-virtualmin-modify-dns/113567) trying to do something very similar (or maybe even exactly what i'm trying to do)

anyway, thanks for your help

**Reply by sebres** [reply]
FWIW, another feature allows now to specify `file://path/to/file-with-ip-set` in `ignoreip`, so fail2ban can read IP, subnet (IP/CIDR), DNS/FQDN or even raw strings to ignore from this file (lazy, fully automatic, without necessity to reload with `fail2ban-client`)
see https://github.com/fail2ban/fail2ban/issues/2013#issuecomment-2695408127 for details.
